### PR TITLE
Add release notes schedule for 1.21

### PR DESCRIPTION
**Pull request by wilsonehusin**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

Add release notes final deadline to schedule

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

The release notes team is expected to iterate on the notes throughout the cycle (roughly every tag cut). This date is the final date we aim to be done by.

/assign @palnabarun

### CODE REVIEW

File: releases/release-1.21/README.md [hunk 1]
```diff
@@ -70,7 +69,9 @@ The 1.21 release cycle is proposed as follows:
 | 1.21.0-rc.0 released | Branch Manager | Tue March 25 | week 11 | |
 | release-1.21 branch created | Branch Manager | Tue March 25 | week 11 | |
 | release-1.21 jobs created | Branch Manager | Tue March 25 | week 11 | |
+| Start final draft of Release Notes | Release Notes Lead | Tue March 25 | week 11 | |
 | Docs complete - All PRs reviewed and ready to merge | Docs Lead | Wed March 31 | week 12 | |
+| Release Notes compelete - reviewed & merged to `k/sig-release` | Release Notes Lead | Mon April 5 | week 13 | |
```
Comment (reylejano) [inline releases/release-1.21/README.md:74]:
```suggestion
| Release Notes complete - reviewed & merged to `k/sig-release` | Release Notes Lead | Mon April 5 | week 13 | |
```

**Review by reylejano — CHANGES_REQUESTED** [review]
Misspelling correction requested for "compelete" to "complete"

File: releases/release-1.21/README.md (hunk 1 above)
Comment (wilsonehusin) [reply releases/release-1.21/README.md:74]: thanks Rey!

**Review by savitharaghunathan — COMMENTED** [review]
/lgtm

@wilsonehusin - Thanks for the PR. Can you squash the commits into one?

**Comment by wilsonehusin** [comment]
done -- thanks @savitharaghunathan!

**Comment by palnabarun** [comment]
@wilsonehusin -- awesome work! Thank you!

/lgtm
/approve
/hold cancel

**Pull request merged**